### PR TITLE
Course theme: update body class when needed

### PR DESCRIPTION
**Pull request by merkushin**
#### Changes proposed in this Pull Request:
- Update the option for the body class when the variation has changed, or styling has been reset.

#### Testing instructions:
- Activate Course theme.
- Choose Gold variation.
- Check 'is-gold' body class exist on the front end.
- Change variation to Blue.
- Check 'is-blue' body class exist on the front end.
- Chang font family in Global Styles. Save.
- Check 'is-blue' body class exist on the front end.
- Reset styling or set default variation.
- Check 'is-defaulit' body class exist on the front end.

### CODE REVIEW

File: course/functions.php [hunk 1]
```diff
@@ -101,14 +101,56 @@ function course_save_global_styles( $post_id, $post, $update ) {
 	$global_styles      = json_decode( $post->post_content, true );
 	$global['settings'] = isset( $global_styles['settings'] ) ? $global_styles['settings'] : array();
 	$global['styles']   = isset( $global_styles['styles'] ) ? $global_styles['styles'] : array();
+
+	if ( empty( $global['settings'] ) && empty( $global['styles'] ) ) {
```
Comment (merkushin) [inline course/functions.php:105]: Is is empty when we reset styling or set default variation

File: course/functions.php [hunk 2]
```diff
@@ -101,14 +101,56 @@ function course_save_global_styles( $post_id, $post, $update ) {
...
+		update_option( 'course_theme_variation', 'default' );
+		return;
+	}
+
+	$td                 = WP_Theme_JSON_Resolver::get_theme_data();
 	$variations         = WP_Theme_JSON_Resolver::get_style_variations();
-	$current_variation  = 'default';
+	$variations[]       = array(
```
Comment (merkushin) [inline course/functions.php:112]: Not really needed as we catch with empty values.
But added it in case behavior changes.

**Comment by Imran92** [comment]
Works well in general for me. In this scenario, it wasn't adding the class as expected 

1. In the editor, click on Browse Styles to go to the variation selection panel
2. Select a Variation, let's say Gold
3. Click on Browse Styles again to come back to the editor
4. Go to Blocks
5. Change the style of a block (for example, Paragraph font size)
6. Save the changes
7. Go to frontend
8. The added class is `is-default`

image

**Comment by merkushin** [comment]
@Imran92 I'm trying to reproduce the issue, but for me it works as expected.
In the screenshot, you can see I have large font size and still have `is-gold` body class.
CleanShot 2023-05-18 at 02 26 35@2x

**Comment by Imran92** [comment]
Hi @merkushin 👋 Here's a video reproducing it. Hopefully this'll help reproducing the issue :) 

https://github.com/Automattic/themes/assets/6820724/5a3a83b3-f59d-4ffc-9bbf-01e4ac4d0b11

**Comment by merkushin** [comment]
I see. You didn't save it after changing the variation.
Yep, that's a case that we can't handle. At least I don't see a solution at the moment.

**Comment by merkushin** [comment]
Can't rely on the body class, I'm closing the PR.

**Pull request closed without merging**